// @react-login-page/page3/cjs/control/Email.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Email = void 0;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _reactLoginPage = require("react-login-page");
var _email = require("../icons/email");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["keyname", "name", "rename", "label"];
var Email = exports.Email = /*#__PURE__*/(0, _react.memo)(function (props) {
  var _props$keyname = props.keyname,
    keyname = _props$keyname === void 0 ? 'email' : _props$keyname,
    name = props.name,
    rename = props.rename,
    _props$label = props.label,
    label = _props$label === void 0 ? 'Email:' : _props$label,
    elmProps = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var _useStore = (0, _reactLoginPage.useStore)(),
    dispatch = _useStore.dispatch;
  if (!elmProps.children) {
    elmProps.children = _email.EmailIcon;
  }
  var nameBase = name || rename || keyname;
  var key = keyname || name;
  (0, _react.useEffect)(function () {
    return dispatch((0, _defineProperty2["default"])({}, "$".concat(key), label));
  }, [label]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactLoginPage.Input, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    type: "email",
    placeholder: "Email",
    index: 1,
    spellCheck: false
  }, elmProps), {}, {
    name: nameBase,
    keyname: key
  }));
});
Email.displayName = 'Login.Email';
