// @react-login-page/page3/cjs/control/Password.js
"use strict";

var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault")["default"];
Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.Password = void 0;
var _objectSpread2 = _interopRequireDefault(require("@babel/runtime/helpers/objectSpread2"));
var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));
var _objectWithoutProperties2 = _interopRequireDefault(require("@babel/runtime/helpers/objectWithoutProperties"));
var _react = require("react");
var _reactLoginPage = require("react-login-page");
var _lock = require("../icons/lock");
var _jsxRuntime = require("react/jsx-runtime");
var _excluded = ["keyname", "name", "rename", "label"];
var Password = exports.Password = /*#__PURE__*/(0, _react.memo)(function (props) {
  var _props$keyname = props.keyname,
    keyname = _props$keyname === void 0 ? 'password' : _props$keyname,
    name = props.name,
    rename = props.rename,
    _props$label = props.label,
    label = _props$label === void 0 ? 'Password:' : _props$label,
    elmProps = (0, _objectWithoutProperties2["default"])(props, _excluded);
  var _useStore = (0, _reactLoginPage.useStore)(),
    dispatch = _useStore.dispatch;
  if (!elmProps.children) {
    elmProps.children = _lock.LockIcon;
  }
  var key = keyname || name;
  (0, _react.useEffect)(function () {
    return dispatch((0, _defineProperty2["default"])({}, "$".concat(key), label));
  }, [label]);
  return /*#__PURE__*/(0, _jsxRuntime.jsx)(_reactLoginPage.Input, (0, _objectSpread2["default"])((0, _objectSpread2["default"])({
    type: "password",
    placeholder: "Password",
    autoComplete: "on",
    index: 2
  }, elmProps), {}, {
    name: name || rename || keyname,
    keyname: key
  }));
});
